indexer/src/main/scala/importers/geonames/PolygonDataFlattener.scala

```diff
@@ -0,0 +1,198 @@
+
+    var featureNames: Seq[DisplayName] = originalFeatureNames
+
+    // some OSM feature names look like A (B), make that into two extra names
+    featureNames ++= originalFeatureNames.flatMap(n => {
+      // If the name looks like "A (B)"
+      parenNameRegex.findFirstIn(n.name).toList.flatMap(_ => {
+        // split it into "A" and "B"
+        List(
+          DisplayName(n.lang, parenNameRegex.replaceAllIn(n.name, "$1")),
+          DisplayName(n.lang, parenNameRegex.replaceAllIn(n.name, "$2"))
+        )
+      })
+    })
+
+    // Additionally, some names are actually multiple comma separated names,
+    // so split those into individual names too
+    featureNames = featureNames.flatMap(n => {
+      n.name.split(",").map(_.trim).filterNot(_.isEmpty).map(namePart =>
+        DisplayName(n.lang, namePart)
+      )
+    })
+
+    // Additionally, some names are actually multiple slash separated names
+    // so split those into individual names too
+    featureNames = featureNames.flatMap(n => {
+      n.name.split("/").map(_.trim).filterNot(_.isEmpty).map(namePart =>
+        DisplayName(n.lang, namePart)
+      )
+    })
+
+
+    featureNames.filterNot(_.name.isEmpty)
+  }
+
+  def processFeatureIterator(
+    defaultNamespace: FeatureNamespace,
+    features: ShapeIterator,
+    polygonMappingConfig: Option[PolygonMappingConfig],
+    outputWriter: Writer
+  ) {
+    val fparts = features.file.getName().split("\\.")
+
+    for {
+      (rawFeature, index) <- features.zipWithIndex
+      feature = new FsqSimpleFeature(rawFeature)
+      geom <- feature.geometry
+    } {
+      if (index % 100 == 0) {
+        logger.info("processing feature %d in %s".format(index, features.file.getName))
+      }
+      
+      val polygonId = getId()
+
+      val source = polygonMappingConfig.flatMap(_.source).getOrElse(features.file.getName())
+      
+      val geomBase64String = Base64.encodeBase64URLSafeString(wkbWriter.write(geom))
+
+      val fidsFromFileName = fparts.lift(0).flatMap(p => Helpers.TryO(p.toInt.toString))
+        .flatMap(i => StoredFeatureId.fromHumanReadableString(i, Some(defaultNamespace))).toList
+
+      val geoidColumnNameToNamespaceMapping =
+        List(
+          "qs_gn_id" -> GeonamesNamespace,
+          "gn_id" -> GeonamesNamespace) ++
+          FeatureNamespace.values.map(namespace => (namespace.name -> namespace))
+      val geoidsFromFile = for {
+        (columnName, namespace) <- geoidColumnNameToNamespaceMapping
+        values <- feature.propMap.get(columnName).toList
+        value <- values.split(",")
+        if value.nonEmpty
+      } yield {
+        // if value already contains ':' it was human-readable to begin with
+        if (value.indexOf(':') != -1) {
+          value
+        } else {
+          "%s:%s".format(namespace.name, value)
+        }
+      }
+      val fidsFromFile = geoidsFromFile
+        .filterNot(_.isEmpty).map(_.replace(".0", ""))
```

Comment:
`.map(_.replace(".0", "")).filterNot(_.isEmpty)` (I know it's not going to happen but this is saner)